spec/models/course/publish_enrichments_spec.rb

```diff
@@ -2,17 +2,17 @@
 
 describe Course, type: :model do
   describe "#enrichments" do
-    subject {
-      create(:course, with_enrichments: [
-               [:published, created_at: 5.days.ago],
-               [:published, created_at: 3.days.ago],
-               [:subsequent_draft, created_at: 1.day.ago],
-             ]).enrichments
-    }
+    let(:first_enrichment) { build(:course_enrichment, :published, created_at: 5.days.ago) }
+    let(:second_enrichment) { build(:course_enrichment, :published, created_at: 3.days.ago) }
+    let(:third_enrichment) { build(:course_enrichment, :subsequent_draft, created_at: 1.day.ago) }
+
+    let(:course) { create(:course, enrichments: [first_enrichment, second_enrichment, third_enrichment]) }
+
+    subject { course.reload.enrichments }
```

Comment:
Looking at the changes to this file in general, I probably would have done something like this:

```
describe Course, type: :model do
  let(:enrichments { [] }
  let(:course)     { create(:course, enrichments: enrichments) }

  subject { course }

  describe "#enrichments" do
    let(:enrichments) do
      [
        build(:course_enrichment, :published, created_at: 5.days.ago)
        build(:course_enrichment, :published, created_at: 3.days.ago)
        build(:course_enrichment, :subsequent_draft, created_at: 1.day.ago)
      ]
    end

    ...
  end

  ...

  context "for a course an initial draft enrichments" do
    let(:enrichments) { [build(:course_enrichment, :initial_draft)]) }
    ...
  end

  ...
```

The idea being that we create the course the same way, and the subject is the same, it's just the `enrichments` that change in the context blocks. Not a blocker though if you want to see the back of these tests.

Reply:
slapped in enrichments